include/seqan3/core/detail/customisation_point.hpp

```diff
@@ -54,3 +56,86 @@ static constexpr decltype(auto) impl(seqan3::detail::priority_tag<PRIO>,
 {                                                                                                                    \
     return TERM;                                                                                                     \
 }
+#endif // SEQAN3_DEPRECATED_310
+
+#if SEQAN3_DOXYGEN_ONLY(1)0
+//!\brief A macro helper for #SEQAN3_CPO_OVERLOAD.
+#define SEQAN3_CPO_OVERLOAD_BODY(...) noexcept(auto) { return __VA_ARGS__; }
+#else // ^^^ (simplified) doxygen version / normal definition vvv
+#define SEQAN3_CPO_OVERLOAD_BODY(...) \
+        noexcept(noexcept(__VA_ARGS__)) \
+        -> decltype(__VA_ARGS__) \
+    { return __VA_ARGS__; } \
+    static_assert(true)
+#endif
+
+/*!\brief A macro that helps to define a seqan3::detail::customisation_point_object.
+ * \details
+ * Expands to a function definition with the name `cpo_overload`, a puts the given expression via
+ * #SEQAN3_CPO_OVERLOAD_BODY as a single return statement in the function body, the noexcept declaration and requires
+ * declaration.
+ *
+ * ```cpp
+ * template <typename alphabet_t>
+ * static constexpr auto SEQAN3_CPO_OVERLOAD(seqan3::detail::priority_tag<2>,
+ *                                           seqan3::alphabet_rank_t<alphabet_t> const rank,
+ *                                           alphabet_t && alphabet)
+ * (
+ *     seqan3::custom::alphabet<alphabet_t>::assign_rank_to(rank, alphabet)
+ * );
+ * ```
```

Comment:
Please move this to a snippet file in `test/snippet/` and include it here.

Reply:
done